pkg/controller/storagecluster/external_resources.go

```diff
@@ -54,39 +55,58 @@ func (r *ReconcileStorageCluster) ensureExternalStorageClusterResources(instance
 	if err != nil {
 		return err
 	}
-	ownerRef := metav1.OwnerReference{
-		UID:        instance.UID,
-		APIVersion: instance.APIVersion,
-		Kind:       instance.Kind,
-		Name:       instance.Name,
-	}
 	var data []ExternalResource
 	err = json.Unmarshal(found.Data[externalClusterDetailsKey], &data)
 	if err != nil {
 		reqLogger.Error(err, "could not parse json blob")
 		return err
 	}
-	scs, err := r.newStorageClasses(instance)
+	err = r.createExternalStorageClusterResources(data, instance, reqLogger)
 	if err != nil {
-		reqLogger.Error(err, "failed to create StorageClasses")
+		reqLogger.Error(err, "could not create ExternalStorageClusterResource")
 		return err
 	}
-	err = r.createExternalStorageClusterResources(ownerRef, scs, data, instance, reqLogger)
-	if err != nil {
-                reqLogger.Error(err, "could not create ExternalStorageClusterResource")
-                return err
-        }
-	// creating all the storageClasses once we set the values
-	err = r.createStorageClasses(scs, reqLogger)
+	instance.Status.ExternalSecretFound = true
+	return nil
+}
+
+func (r *ReconcileStorageCluster) setRookCSICephFS(
```

Comment:
Added the function from @bipuladh 's PR: https://github.com/openshift/ocs-operator/pull/697

Reply:
Done...